src/Metrics/TargetDetectionStatusesProvider.php

```diff
@@ -53,7 +53,7 @@ public function __construct(
         private readonly Logs $logConfig,
         private readonly string $logVerbosity,
         private readonly bool $onlyCoveredMode,
-        private readonly bool $showMutations,
+        private readonly ?int $numberOfShownMutations,
```

Comment:
```suggestion
        private readonly ?int $numberOfShownMutations = null,
```
Sane defaults could reduce a whole lot of boilerplate code in tests.

Reply:
I am not a fan of defaults for such cases, as it might easily lead to application code which forgets to pass this value down the line